### PR TITLE
chore: Periodic merge of main into v2

### PR DESCRIPTION
**Pull request by scottgerring**
**Issue #, if available:**

## Description of changes:

**Checklist**

 

* [ ] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/run-e2e-tests.yml [hunk 1]
```diff
@@ -33,13 +33,19 @@ jobs:
...
     name: End-to-end tests java${{ matrix.java }}
     env:
-      JAVA_VERSION: ${{ matrix.java }}
       AWS_DEFAULT_REGION: eu-west-1
+
+      # If matrix.version is 21, use 17, otherwise use matrix.version
+      # This is because AspectJ does not yet support weaving with Java21; we want
+      # to test the Java21 runtime, but we can't yet use the JDK21 compiler.
+      # https://github.com/eclipse-aspectj/aspectj/issues/260#issuecomment-1815920274
+      JAVA_VERSION: ${{ (matrix.java == 21 && '17') || matrix.java }}
```
Comment (jeromevdl) [inline .github/workflows/run-e2e-tests.yml:47]: we should remove that...

File: .github/workflows/run-e2e-tests.yml [hunk 2]
```diff
@@ -49,7 +55,8 @@ jobs:
         uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3.11.0
         with:
           distribution: 'corretto'
-          java-version: ${{ matrix.java }}
+          # See comment above on JAVA_VERSION env var
+          java-version: ${{ (matrix.java == 21 && '17') || matrix.java }}
```
Comment (jeromevdl) [inline .github/workflows/run-e2e-tests.yml:59]: same here

File: README.md [hunk 3]
```diff
@@ -11,6 +11,14 @@ Powertools for AWS Lambda (Java) is a developer toolkit to implement Serverless
 
 **[📜Documentation](https://docs.powertools.aws.dev/lambda-java/)** | **[Feature request](https://github.com/aws-powertools/powertools-lambda-java/issues/new?assignees=&labels=feature-request%2C+triage&template=feature_request.md&title=)** | **[🐛Bug Report](https://github.com/aws-powertools/powertools-lambda-java/issues/new?assignees=&labels=bug%2C+triage&template=bug_report.md&title=)** | **[Detailed blog post](https://aws.amazon.com/blogs/opensource/simplifying-serverless-best-practices-with-aws-lambda-powertools-java/)**
 
+### Java Compatibility
```
Comment (jeromevdl) [inline README.md:14]: this is not from main, but a previous version...

**Review by jeromevdl — CHANGES_REQUESTED** [review]
not the latest version (README)

File: .github/workflows/run-e2e-tests.yml (hunk 1 above)
Comment (scottgerring) [reply .github/workflows/run-e2e-tests.yml:47]: We should remove it in main, this is just a branch merge

File: powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/LambdaTimestampResolver.java
Comment (jeromevdl) [inline powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/LambdaTimestampResolver.java:1]: this file should not be ported to v2, ca we do that ?

File: powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/LambdaTimestampResolverFactory.java
Comment (jeromevdl) [inline powertools-logging/src/main/java/software/amazon/lambda/powertools/logging/internal/LambdaTimestampResolverFactory.java:1]: same here...

File: powertools-logging/src/main/resources/LambdaJsonLayout.json
Comment (jeromevdl) [inline powertools-logging/src/main/resources/LambdaJsonLayout.json:1]: this should not be taken

**Review by jeromevdl — APPROVED** [review]
Apart from the logging (ALC), it's ok, should I do the appropriate merge in the logging branch?

**Pull request merged**